rag.py

```
@@ -371,6 +351,131 @@ def llm_generate(prompt: str, max_tokens: int = None,
+      chunks_used, prompt, metadatas, detected_lang, lang_name
+    If no docs: chunks_used=0, no_docs_msg set instead of prompt/metadatas.
+    """
+    detected_lang = lang_utils.detect_language(user_query) or "en"
+    lang_name = lang_utils.get_language_name(detected_lang)
+
+    retrieval_query = user_query
+    if strategy == "B" and detected_lang != "en" and lang_utils.is_indic_language(detected_lang):
+        try:
+            retrieval_query = translation.translate_to_english(user_query, detected_lang)
+        except Exception:
+            pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)